src/System.Drawing.Common/tests/mono/System.Drawing/BitmapTests.cs

```diff
@@ -0,0 +1,1596 @@
+        }
+
+        static string ByteArrayToString(byte[] arrInput)
+        {
+            int i;
+            StringBuilder sOutput = new StringBuilder(arrInput.Length);
+            for (i = 0; i < arrInput.Length - 1; i++)
+            {
+                sOutput.Append(arrInput[i].ToString("X2"));
+            }
+            return sOutput.ToString();
+        }
+
+
+        public string RotateBmp(Bitmap src, RotateFlipType rotate)
+        {
+            int width = 150, height = 150, index = 0;
+            byte[] pixels = new byte[width * height * 3];
+            Bitmap bmp_rotate;
+            byte[] hash;
+            Color clr;
+
+            bmp_rotate = src.Clone(new RectangleF(0, 0, width, height), PixelFormat.Format32bppArgb);
+            bmp_rotate.RotateFlip(rotate);
+
+            for (int y = 0; y < height; y++)
+            {
+                for (int x = 0; x < width; x++)
+                {
+                    clr = bmp_rotate.GetPixel(x, y);
+                    pixels[index++] = clr.R;
+                    pixels[index++] = clr.G;
+                    pixels[index++] = clr.B;
+                }
+            }
+
+            hash = MD5.Create().ComputeHash(pixels);
+            return ByteArrayToString(hash);
+        }
+        public string RotateIndexedBmp(Bitmap src, RotateFlipType type)
+        {
+            int pixels_per_byte;
+
+            switch (src.PixelFormat)
+            {
+                case PixelFormat.Format1bppIndexed:
+                    pixels_per_byte = 8;
+                    break;
+                case PixelFormat.Format4bppIndexed:
+                    pixels_per_byte = 2;
+                    break;
+                case PixelFormat.Format8bppIndexed:
+                    pixels_per_byte = 1;
+                    break;
+
+                default:
+                    throw new Exception("Cannot pass a bitmap of format " + src.PixelFormat + " to RotateIndexedBmp");
+            }
+
+            Bitmap test = src.Clone() as Bitmap;
+
+            test.RotateFlip(type);
+
+            BitmapData data = null;
+            byte[] pixel_data;
+
+            try
+            {
+                data = test.LockBits(new Rectangle(0, 0, test.Width, test.Height), ImageLockMode.ReadOnly, test.PixelFormat);
+
+                int scan_size = (data.Width + pixels_per_byte - 1) / pixels_per_byte;
+                pixel_data = new byte[data.Height * scan_size];
+
+                for (int y = 0; y < data.Height; y++)
+                {
+                    IntPtr src_ptr = (IntPtr)(y * data.Stride + data.Scan0.ToInt64());
+                    int dest_offset = y * scan_size;
+                    for (int x = 0; x < scan_size; x++)
+                        pixel_data[dest_offset + x] = Marshal.ReadByte(src_ptr, x);
+                }
+            }
+            finally
+            {
+                if (test != null)
+                {
+                    if (data != null)
```

Comment:
Could you please add braces and use multiple lines in the try blocks in here? Would look much cleaner